docs/Validation.md

```diff
@@ -0,0 +1,252 @@
+  .every.number()
+  .test([1, 2, "hello"]); // False, hello isn't a number
+
+v8n()
+  .some.number()
+  .test([1, 2, "hello"]); // True, there is at least one number
+```
+
+::: tip
+Modifiers can be combined on a rule. Simply chain them!
+:::
+
+Notice how modifiers don't have braces `()`, this distinguishes them from
+regular rules. There are three modifiers provided which you can use. Two of them
+cater to arrays and one is more general purpose. Take a peek at all the
+[built-in modifiers](/api/#built-in-modifiers) for a little more info.
+
+## Validation strategies
+
+Most validation will only return `true` or `false`. While v8n offers this, it
+also gives you the ability to validate with different output formats. There are
+currently 4 validation strategies for use and all of them are slightly different
+from one another.
+
+The validation strategy is chosen with the method at the end of your validation
+chain. This method also receives the value you want to validate as it's only
+argument. So keep in mind that whatever you actually want to validate is written
+at the end of the chain, not anywhere within the rules or at the beginning.
+
+### Boolean based validation
+
+Of course when validating many people will want to know if the given value
+passes or not. This is regular `boolean` validation and v8n performs this if
+you use the `test()` method. Quite simply this will return true if all the rules
+pass and false if any of them fails.
+
+```js
+v8n()
+  .string()
+  .test("Melon"); // True
+
+v8n()
+  .string()
+  .test({ age: 3 }); // False, this is an object
+```
+
+This strategy is most useful for conditionals since you can pass it's result
+directly into an `if` statement or similar. For more details about `test()`
+and for additional examples take a look at it's [API documentation](/api/#test).
+
+### Array based validation
+
+If you really need to know what was wrong with the value you passed to v8n,
+simply getting `true` or `false` won't do you any good. This is where
+array-based comes in. If you use `testAll()` you will always receive an array
+in return. The array will be empty if no rules failed, but it will contain all
+the rules that failed if any do. This also means that this strategy does not
+stop when any rule fails like the boolean-based version does, instead it will
+always run all the rules.
+
+```js
+v8n()
+  .string()
+  .first("H")
+  .last("o")
+  .testAll("Hello"); // Returns [] since every rule passes
+
+v8n()
+  .string()
+  .first("H")
+  .last("o")
+  .testAll("Hi"); // Returns [Rule{name: "last"...}] (last letter isn't o)
+```
+
+This is useful for providing detailed error messages but can also be used for
+any number of other purposes. For some more in-depth examples head over
+[to the documentation for `testAll()`](/api/#testall).
+
+::: tip
+The array will contain [`Rule` objects](/api/#rule) that you can work with.
+:::
+
+Keep in mind that this validation strategy will work exactly opposite if used
```

Comment:
This information seems to be important for the user, so it should be inside **warning** block.